src/concrete/ml/pandas/_operators.py

```diff
@@ -0,0 +1,241 @@
+import numpy
+import pandas
+from pandas.core.reshape.merge import _MergeOperation
+
+
+def check_coherence_selected_column_for_merge(df_left, df_right, on):
+    on_left, on_right = df_left.dtype_mappings[on], df_right.dtype_mappings[on]
+    dtype_left, dtype_right = numpy.dtype(on_left["dtype"]), numpy.dtype(on_right["dtype"])
+
+    if dtype_left == dtype_right:
+        if numpy.issubdtype(dtype_left, numpy.floating):
+            raise ValueError(
+                f"Column '{on}' cannot be selected for merging both data-frames because it has a "
+                f"floating dtype ({dtype_left})"
+            )
+        elif dtype_left == "object":
+            str_mapping_left = on_left["str_to_int"]
+            str_mapping_right = on_right["str_to_int"]
+
+            # TODO: add hash
+            if str_mapping_left != str_mapping_right:
+                raise ValueError(
+                    f"Mappings for string values in both data-frames' common column '{on}' do not "
+                    "match."
+                )
+    else:
+        raise ValueError(
+            f"Dtype of both data-frames' common column '{on}' do not match. Got {dtype_left} "
+            f"(left) and {dtype_right} (right)."
+        )
+
+
+def encrypted_left_join(
+    df_left,
+    df_right,
+    server,
+    on=None,
+    left_on=None,
+    right_on=None,
+    left_index=False,
+    right_index=False,
+    sort=False,
+    copy=None,
+    indicator=False,
+    validate="1:1",
+):
```

Comment:
What's the output data type? Pandas, Numpy? We can probably add some typing here. Same for the args.